### PR TITLE
(Re)implement track rendering

### PR DESCRIPTION
**Pull request by mwoehlke-kitware**
Add detections/states to `KwiverTrackModel`, so that we have a way with the new data model framework to get bounding boxes. Add `ScalarFilterModel`, to filter a model based on scalar bounds, which will be used to determine what boxes should be shown on a given frame. Update detection rendering code to use these.

### CODE REVIEW

File: sealtk/core/test/ScalarFilterModel.cpp [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+      case sealtk::core::StartTimeRole:
+      case sealtk::core::EndTimeRole:
+        return QVariant::fromValue(static_cast<time_us_t>(row * 100));
+
+      default:
+        break;
+    }
+  }
+
+  return sealtk::core::AbstractItemModel::data(index, role);
```
Comment (KyleFromKitware) [inline sealtk/core/test/ScalarFilterModel.cpp]: `this->`

File: sealtk/core/test/ScalarFilterModel.cpp [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+#include <QSet>
+
+#include <QtTest>
+
+namespace kvr = kwiver::vital::range;
+
+using time_us_t = kwiver::vital::timestamp::time_t;
+
+// ----------------------------------------------------------------------------
+template<> char* QTest::toString<QSet<int>>(QSet<int> const& set)
```
Comment (KyleFromKitware) [inline sealtk/core/test/ScalarFilterModel.cpp]: `error: specialization of 'template<class T> char* QTest::toString(const T&)' in different namespace ... from definition of 'template<class T> char* QTest::toString(const T&)' [-fpermissive]`

Comment (KyleFromKitware) [reply sealtk/core/test/ScalarFilterModel.cpp]: Wrapping it in `namespace QTest{` and removing `QTest::` fixes the error for me.

Comment (KyleFromKitware) [reply sealtk/core/test/ScalarFilterModel.cpp]: Looks like this is a [GCC bug](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=56480). Still, it's an easy fix, so we should fix it anyway.

**Comment by KyleFromKitware** [comment]
@mwoehlke-kitware I just pushed an update to fix the issues I mentioned.

File: sealtk/core/test/ScalarFilterModel.cpp (hunk 1 above)
Comment (mwoehlke-kitware) [reply sealtk/core/test/ScalarFilterModel.cpp]: What is the actual rationale for this? I see *one* place that follows this pattern, currently, and several that don't.

**Pull request merged**